ibis/expr/types/relations.py

```diff
@@ -2417,12 +2441,31 @@ def drop(self, *fields: str | Selector) -> Table:
         │ Torgersen │               193 │        3450 │ female │  2007 │
         └───────────┴───────────────────┴─────────────┴────────┴───────┘
         """
-        if not fields:
+
+        def _get_cols(field) -> list[str]:
+            try:
+                cols = self._fast_bind(field)
+            except com.IbisTypeError as e:
+                if "is not found in table" in str(e):
+                    if missing_ok:
+                        return []
+                    raise
+            except AttributeError as e:
+                if "has no attribute" in str(e):
```

Comment:
Checking for substrings in error messages is extremely fragile. The implementation of this functionality should basically never depend on the specific content an informational error message.

Reply:
Additionally, binding for every column is very expensive for very wide tables.

Instead, what we should do is the following:

1. Introduce `FieldNotFoundError` as you say
2. Allow bind (perhaps through an option to `_fast_bind`?) to return an optimistic (possibly) subset: basically whatever columns are found from `fields`.